src/os.cc

```diff
@@ -366,8 +366,32 @@ long getpagesize() {
 #  endif
 }
 
-FMT_API void ostream::grow(size_t) {
+FMT_BEGIN_DETAIL_NAMESPACE
+
+void file_buffer::grow(size_t) {
   if (this->size() == this->capacity()) flush();
 }
+
+file_buffer::file_buffer(cstring_view path,
+                         const detail::ostream_params& params)
+    : file_(path, params.oflag) {
+  set(new char[params.buffer_size], params.buffer_size);
+}
+
+file_buffer::file_buffer(file_buffer&& other)
+    : detail::buffer<char>(other.data(), other.size(), other.capacity()),
+      file_(std::move(other.file_)) {
+  other.clear();
+  other.set(nullptr, 0);
+}
+
+file_buffer::~file_buffer() {
+  flush();
+  delete[] data();
+}
+
+FMT_END_DETAIL_NAMESPACE
+
+ostream::~ostream() = default;
```

Comment:
Does it need to be outlined?

Reply:
it guess that's my old habit - always anchor class by its d-tor
otherwise - how would compiler know where to place definition of functions marked with `FMT_API`?

maybe that is not the case for this particular class - but that's the case for polymorphic classes
at least I didn't check if it would behave fine w/o it